src/main/java/bio/terra/service/resourcemanagement/google/GoogleResourceService.java

```diff
@@ -572,6 +586,55 @@ public void enableIamPermissions(Map<String, List<String>> userPermissions, Stri
         throw new EnablePermissionsFailedException("Cannot enable iam permissions", lastException);
     }
 
+    public void removeIamPermissions(Map<String, List<String>> userPermissions, String projectId)
+        throws InterruptedException {
+
+        GetIamPolicyRequest getIamPolicyRequest = new GetIamPolicyRequest();
+
+        Exception lastException = null;
+        int retryWait = INITIAL_WAIT_SECONDS;
+        for (int i = 0; i < RETRIES; i++) {
+            try {
+                CloudResourceManager resourceManager = cloudResourceManager();
+                Policy policy = resourceManager.projects()
+                    .getIamPolicy(projectId, getIamPolicyRequest).execute();
+                final List<Binding> bindingsList = policy.getBindings();
+
+                // Remove members from the current policies
+                for (Map.Entry<String, List<String>> entry : userPermissions.entrySet()) {
+                    CollectionUtils.transform(bindingsList, b -> {
+                        if (Objects.equals(b.getRole(), entry.getKey())) {
+                            // Remove any deleted members explicitly
+                            b.setMembers(ListUtils.select(b.getMembers(), m -> !m.startsWith("deleted:")));
+                            // Remove the members that were passed in
+                            b.setMembers(ListUtils.subtract(b.getMembers(), entry.getValue()));
+                        }
+                        return b;
+                    });
+                }
+
+                // Remove any entries from the bindings list with no members
+                CollectionUtils.filter(bindingsList, b -> !b.getMembers().isEmpty());
```

Comment:
So, `CollectionUtils.filter` side effects its argument but `ListUtils.select` does not. It seems to me that you could get rid of the `setMembers` calls above by using `CollectionUtils.filter` instead. I'm not too familiar with these utility functions; I would've guessed that they both either modified or didn't.

Reply:
This is just a safety latch after to make sure that after having removed the last member in a binding, we remove that entry altogether.  Are you suggesting that the above call should be a filter that side-effects to also remove an entry from the list?

Reply:
I don't have a strong opinion about one way or the other, my point is that since `x = ListUtils.select(x, ...)` does the same thing as `CollectionUtils.filter(x, ...)` it seemed odd to use one here and another above. E.g. either I'd change this to be
```
bindingsList = ListUtils.select(bindingsList, b -> !b.getMembers().isEmpty());
```
Or change the code above to use `CollectionUtils.filter()`.

Reply:
(My personal preference would be to not use a library at all and write:
```
policy.setBindings(bindingsList.stream().filter(b -> !b.getMembers().isEmpty()).collect(Collectors.toList()));
```
)

Reply:
I changed it to use filter and perform the side effect